sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -418,6 +433,36 @@ class SQLQueryTestSuite extends QueryTest with SharedSparkSession with SQLHelper
     }
   }
 
+  def expandCTEQueryAndCompareResult(
+      session: SparkSession,
+      query: String,
+      output: ExecutionOutput): Unit = {
+    val logicalPlan: LogicalPlan = session.sessionState.sqlParser.parsePlan(query)
+    // For non-command query with CTE, compare the results of selecting from view created on the
+    // original query.
```

Comment:
note: this may not always work. CREATE VIEW requires unique column names but a query can output columns with the same name. We can add one more condition here and only trigger the view test if the output schema has no duplicated names.